### DISCUSSION TITLE
If my code has multiple targets, how could I make them both active in rust-analyzer

### DISCUSSION THREAD

**Discussion by Dumpinground** [rust-lang/rust-analyzer#15293] [Q&A] [answered]
For now the wasm32 code is inactive.
![image](https://github.com/rust-lang/rust-analyzer/assets/56429201/13d15311-f361-47e2-8983-37ccb3bb437f)
I tried `rust-analyzer.cargo.target` but it seems only one target can be set.
Also I tried `rust-analyzer.check.targets`. It is effective for `cargo check` but not `rust-analyzer`.
Did I miss something?

**Comment by flodiebold** [comment, marked as answer]
You can't. rust-analyzer can currently only analyze one active target. In principle having multiple targets/configurations active at the same time is a consideration, but it's 1. complicated to get right (all the IDE features have to choose which configuration to use, or whether they have to consider all of them), and 2. would impact performance quite a bit.

**Reply by Dumpinground** [reply]
Ok, I get it. Thanks for your answer
